packages/hebao_v1/contracts/stores/SecurityStore.sol

```diff
@@ -53,46 +55,93 @@ contract SecurityStore is DataStore
         view
         returns (bool)
     {
-        return wallets[wallet].guardianIdx[guardian] > 0;
+        return wallets[wallet].guardianIdx[guardian] > 0 &&
+            wallets[wallet].guardianEffectiveTime[guardian] >= now &&
+            wallets[wallet].guardianRemovalEffectiveTime[guardian] < now;
     }
 
     function getGuardian(
         address wallet,
-        address _guardian
+        address guardian
         )
         public
         view
-        returns (Data.Guardian memory guardian)
+        returns (Data.Guardian memory)
     {
-        uint index = wallets[wallet].guardianIdx[_guardian];
+        Wallet storage w = wallets[wallet];
+        uint index = w.guardianIdx[guardian];
         require(index > 0, "NOT_A_GUARDIAN");
-        guardian = wallets[wallet].guardians[index-1];
+        require(
+            w.guardianEffectiveTime[guardian] >= now &&
+            w.guardianRemovalEffectiveTime[guardian] < now,
```

Comment:
Maybe best to create a `isGuardianInternal` that contains this logic as it is duplicated at a couple of places.